opentelemetry-sdk/src/opentelemetry/sdk/metrics/_internal/exponential_histogram/mapping/ieee_754.py

```diff
@@ -0,0 +1,125 @@
+# Copyright The OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from ctypes import c_double, c_uint64
+from decimal import Decimal
+from sys import float_info
+
+# An IEEE 754 double-precision (64 bit) floating point number is represented
+# as: 1 bit for sign, 11 bits for exponent and 52 bits for significand. Since
+# these numbers are in a normalized form (in scientific notation), the first
+# bit of the significand will always be 1. Because of that, that bit is not
+# stored but implicit, to make room for one more bit and more precision.
+
+SIGNIFICAND_WIDTH = 52
+EXPONENT_WIDTH = 11
+
+# This mask is equivalent to 52 "1" bits (there are 13 hexadecimal 4-bit "f"s
+# in the significand mask, 13 * 4 == 52) or 0xfffffffffffff in hexadecimal.
+SIGNIFICAND_MASK = (1 << SIGNIFICAND_WIDTH) - 1
+
+# There are 11 bits for the exponent, but the exponent bias values 0 (11 "0"
+# bits) and 2047 (11 "1" bits) have special meanings so the exponent range is
+# from 1 to 2046. To calculate the exponent value, 1023 is subtracted from the
+# exponent, so the exponent value range is from -1022 to +1023.
+EXPONENT_BIAS = (2 ** (EXPONENT_WIDTH - 1)) - 1
+
+# All the exponent mask bits are set to 1 for the 11 exponent bits.
+EXPONENT_MASK = ((1 << EXPONENT_WIDTH) - 1) << SIGNIFICAND_WIDTH
+
+# The exponent mask bit is to 1 for the sign bit.
+SIGN_MASK = 1 << (EXPONENT_WIDTH + SIGNIFICAND_WIDTH)
+
+MIN_NORMAL_EXPONENT = -EXPONENT_BIAS + 1
+MAX_NORMAL_EXPONENT = EXPONENT_BIAS
+
+# Smallest possible normal value (2.2250738585072014e-308)
+# This value is the result of using the smallest possible number in the
+# mantissa, 1.0000000000000000000000000000000000000000000000000000 (52 "0"s in
+# the fractional part) = 1.0000000000000000 and a single "1" in the exponent.
+# Finally 1.0000000000000000 * 2 ** -1022 = 2.2250738585072014e-308.
+MIN_NORMAL_VALUE = float_info.min
```

Comment:
Good question, I think it is guaranteed because the [documentation](https://docs.python.org/3/library/sys.html#sys.float_info) says:

> The values correspond to the various floating-point constants defined in the standard header file float.h for the ‘C’ programming language; see section 5.2.4.2.2 of the 1999 ISO/IEC C standard [C99], ‘Characteristics of floating types’, for details.

Now, [5.2.4.2.2 of the 1999 ISO/IEC C standard [C99]](https://www.open-std.org/jtc1/sc22/wg14/www/docs/n1256.pdf) says, in its 14th point, Example 2:

> DBL_MIN 2.2250738585072014E-308

Which is the same value I get when I do this:

```
Python 3.10.4 (main, Oct  3 2022, 16:16:22) [GCC 11.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from sys import float_info
>>> float_info.min
2.2250738585072014e-308
```